src/nodedef.cpp

```diff
@@ -51,16 +52,24 @@ void NodeBox::serialize(std::ostream &os, u16 protocol_version) const
 	writeU8(os, 6); // version. Protocol >= 36
 
 	switch (type) {
-	case NODEBOX_LEVELED:
 	case NODEBOX_FIXED:
+	case NODEBOX_LEVELED: {
 		writeU8(os, type);
 
 		writeU16(os, fixed.size());
 		for (const aabb3f &nodebox : fixed) {
 			writeV3F32(os, nodebox.MinEdge);
 			writeV3F32(os, nodebox.MaxEdge);
 		}
+		if (type == NODEBOX_LEVELED) {
+			writeU16(os, leveled_fixed.size());
+			for (const aabb3f &nodebox : leveled_fixed) {
+				writeV3F32(os, nodebox.MinEdge);
+				writeV3F32(os, nodebox.MaxEdge);
+			}
+		}
```

Comment:
did you check the protocol cross-compatibility?
hint: it's broken by this. `ContentFeatures::serialize` uses node boxes "bare" so serializing more bytes will confuse old clients and vice-versa